src/replica/replica_stub.cpp

```diff
@@ -362,6 +387,35 @@ void replica_stub::initialize(bool clear /* = false*/)
     _access_controller = std::make_unique<dsn::security::access_controller>();
 }
 
+dsn::error_s store_kms_key(const std::string &data_dir,
+                           const std::string &encryption_key,
+                           const std::string &iv,
+                           const std::string &key_version)
+{
+    replica_kms_info kms_info(encryption_key, iv, key_version);
+    auto err = kms_info.store(data_dir);
+    if (err != dsn::ERR_OK) {
+        return dsn::error_s::make(err, "Can't open kms-info file to write");
+    }
+    return dsn::error_s::ok();
+}
+
+void get_kms_key(const std::string &data_dir,
```

Comment:
It would be better to return error_s as well, distinguish errors like "not found" and other type of errors. Then judge the return value when call this function.